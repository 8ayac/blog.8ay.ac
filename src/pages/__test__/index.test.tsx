import TopPage from '@/src/pages';
import { mount } from 'enzyme';
import React from 'react';

describe('TopPage', () => {
  const testCategories = ['test-cat01', 'test-cat02', 'test-cat03'];
  const testArticles = [
    {
      id: 'test-article-01',
      title: 'Test Title 01',
      tags: testCategories.slice(0, 3),
      publishedAt: new Date('2021-01-01T00:00:00.000Z'),
      updatedAt: new Date('2021-01-01T00:00:00.000Z'),
      body: 'bluhbluhbluh',
    },
    {
      id: 'test-article-02',
      title: 'Test Title 02',
      tags: testCategories.slice(0, 2),
      publishedAt: new Date('2021-02-02T00:00:00.000Z'),
      updatedAt: new Date('2021-02-02T00:00:00.000Z'),
      body: 'bluhbluhbluh',
    },
    {
      id: 'test-article-03',
      title: 'Test Title 03',
      tags: testCategories.slice(0, 1),
      publishedAt: new Date('2021-03-03T00:00:00.000Z'),
      updatedAt: new Date('2021-03-03T00:00:00.000Z'),
      body: 'bluhbluhbluh',
    },
  ];

  test('is rendered correctly to match the snapshot', () => {
    const wrapper = mount(<TopPage articles={testArticles} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('has proper style rules', () => {
    const wrapper = mount(<TopPage articles={testArticles} />);

    test('in ArticleListSection', () => {
      expect(wrapper.find('S-ArticleListSection')).toHaveStyleRule(
        'margin-bottom',
        '4rem'
      );
    });

    test('in TagListSection', () => {
      expect(wrapper.find('S-TagListSection')).toHaveStyleRule(
        'margin-bottom',
        '4rem'
      );
    });
  });
});
