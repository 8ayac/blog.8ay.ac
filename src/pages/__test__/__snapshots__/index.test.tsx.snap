// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopPage is rendered correctly to match the snapshot 1`] = `
<Fragment>
  <S-ArticleListSection>
    <BodyHeader
      title="Articles"
    />
    <ArticleHeaderList
      articles={
        Array [
          Object {
            "body": "bluhbluhbluh",
            "id": "test-article-01",
            "publishedAt": "2021-01-01 00:00:00 +0000",
            "tags": Array [
              "test-cat01",
              "test-cat02",
              "test-cat03",
            ],
            "title": "Test Title 01",
            "updatedAt": "2021-01-01 00:00:00 +0000",
          },
          Object {
            "body": "bluhbluhbluh",
            "id": "test-article-02",
            "publishedAt": "2021-02-02 00:00:00 +0000",
            "tags": Array [
              "test-cat01",
              "test-cat02",
            ],
            "title": "Test Title 02",
            "updatedAt": "2021-02-02 00:00:00 +0000",
          },
          Object {
            "body": "bluhbluhbluh",
            "id": "test-article-03",
            "publishedAt": "2021-03-03 00:00:00 +0000",
            "tags": Array [
              "test-cat01",
            ],
            "title": "Test Title 03",
            "updatedAt": "2021-03-03 00:00:00 +0000",
          },
        ]
      }
    />
  </S-ArticleListSection>
  <S-CategoryListSection>
    <BodyHeader
      title="Tags"
    />
    <CategoryTagList
      tags={
        Array [
          "test-cat01 (3)",
          "test-cat02 (2)",
          "test-cat03 (1)",
          "test-cat01 (3)",
          "test-cat02 (2)",
          "test-cat01 (3)",
        ]
      }
    />
  </S-CategoryListSection>
</Fragment>
`;
